contracts/DePayRouterV1CurveFiSwap01.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.5 <0.8.0;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import './interfaces/ICurveFiSwap.sol';
+import './libraries/Helper.sol';
+
+contract DePayRouterV1CurveFiSwap01 {
+  
+  using SafeMath for uint;
+
+  // Address representating ETH (e.g. in payment routing paths)
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+  // MAXINT to be used only, to increase allowance from
+  // payment protocol contract towards known 
+  // decentralized exchanges, not to dyanmically called contracts!!!
+  uint public immutable MAXINT = type(uint256).max;
+  
+  // Address of Synth sETH (sETH).
+  // They're mostly using sETH
+  // https://etherscan.io/address/0x5e74C9036fb86BD7eCdcb084a0673EFc32eA31cb
+  address public immutable SETH;
+
+  // Address of CurveFiSwap
+  address public immutable CurveFiSwap;
+
+  // Indicates that this plugin requires delegate call
+  bool public immutable delegate = true;
+
+  // Pass SETH and the _CurveFiSwap when deploying this contract.
+  constructor ( address _SETH, address _CurveFiSwap) {
+    SETH = _SETH;
+    CurveFiSwap = _CurveFiSwap;
+  }
+
+  // Swap tokenA<>tokenB, ETH<>sETH or sETH<>ETH on CureFi.
+  //
+  // path -> [from, pool, to]
+  // amounts -> [amount, expected]
+  // addresses -> [receiver]
+  //
+  //  function exchange(
+  //      address _pool,      # Pool address, could able to get from 
+  //      address _from,      # From token address
+  //      address _to,        # To token address
+  //      uint256 _amount,    # Amount
+  //      uint256 _expected,  # Minimum amount of from token, like you expect to exchange for 1 USDT
+  //      address _receiver,  # Receiver address
+  //  ) payable returns (uint256);
+  //
+  function execute(
+    address[] calldata path,
+    uint[] calldata amounts,
+    address[] calldata addresses,
+    string[] calldata data
+  ) external payable returns(bool) {
+    // If there are an record in addresses we will use it as receiver address, otherwise
+    address receiver = addresses.length == 1 ? addresses[0] : address(this);
```

Comment:
I've just found CurveFi allows use to move their fund more freely. To me, I'd love optional parameters and universal interface. It would be adaptable and flexible.

In our case, we won't be received any benefit from it.

Reply:
It might create more security issues.

Reply:
btw. we can always create more plugins if we want to, that allow more flexibility like swapping the funds directly back to the user

e.g. `DePayRouterV1CurveFiDirectSwap01`

I would just want to have the basic 01 versions of all swap plugins to all be simple and symmetrical (do the same thing the same way).